lib/omniauth/strategies/openid_connect.rb

```diff
@@ -372,16 +372,18 @@ def id_token_callback_phase
       end
 
       def valid_response_type?
-        return true if params.key?(configured_response_type)
+        return true if configured_response_types.all? { |key| params[key].present? }
```

Comment:
In looking at https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#MultiValueResponseTypes and https://www.rfc-editor.org/rfc/rfc6749 closely, is this the correct mapping?

|`response_type`|Expected parameters|
|---|---|
|`code token`|`access_token`, `token_type`, `code`
|`code id_token`|`code`, `id_token`|
|`id_token token`|`access_token`, `token_type`, `id_token`|
|`code id_token token`|`code`, `id_token`, `token_type`|

Reply:
So while this change is correct for `code id_token`, I think we probably want to handle these cases here. It looks to me that every time `token` shows up, `access_token` and `token_type` come along.

Reply:
Thank you for laying this out @stanhu. I don't have the bandwidth to round out the other combinations, but this changeset covers our use case.

Could we merge this change, then subsequently cover the rest?

If not, feel free to take this over.

Thank you!